src/test/javascript/spec/component/lecture-unit/online-unit/create-online-unit.component.spec.ts

```diff
@@ -0,0 +1,119 @@
+                                        switch (key) {
+                                            case 'lectureId':
+                                                return 1;
+                                        }
+                                    },
+                                }),
+                                parent: {
+                                    paramMap: of({
+                                        get: (key: string) => {
+                                            switch (key) {
+                                                case 'courseId':
+                                                    return 1;
+                                            }
+                                        },
+                                    }),
+                                },
+                            },
+                        },
+                    },
+                },
+            ],
+            schemas: [],
+        })
+            .compileComponents()
+            .then(() => {
+                createOnlineUnitComponentFixture = TestBed.createComponent(CreateOnlineUnitComponent);
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should send POST request upon form submission and navigate', fakeAsync(() => {
+        const router: Router = TestBed.inject(Router);
+        const onlineUnitService = TestBed.inject(OnlineUnitService);
+
+        const formDate: OnlineUnitFormData = {
+            name: 'Test',
+            releaseDate: dayjs().year(2010).month(3).date(5),
+            description: 'Lorem Ipsum',
+            source: 'https://www.example.com',
+        };
+
+        const response: HttpResponse<OnlineUnit> = new HttpResponse({
+            body: new OnlineUnit(),
+            status: 201,
+        });
+
+        const createStub = jest.spyOn(onlineUnitService, 'create').mockReturnValue(of(response));
+        const navigateSpy = jest.spyOn(router, 'navigate');
+
+        createOnlineUnitComponentFixture.detectChanges();
+        tick();
+        const onlineUnitForm: OnlineUnitFormStubComponent = createOnlineUnitComponentFixture.debugElement.query(By.directive(OnlineUnitFormStubComponent)).componentInstance;
+        onlineUnitForm.formSubmitted.emit(formDate);
+
+        createOnlineUnitComponentFixture.whenStable().then(() => {
+            const onlineUnitCallArgument: OnlineUnit = createStub.mock.calls[0][0];
+            const lectureIdCallArgument: number = createStub.mock.calls[0][1];
+
+            expect(onlineUnitCallArgument.name).toEqual(formDate.name);
+            expect(onlineUnitCallArgument.description).toEqual(formDate.description);
+            expect(onlineUnitCallArgument.releaseDate).toEqual(formDate.releaseDate);
+            expect(onlineUnitCallArgument.source).toEqual(formDate.source);
+            expect(lectureIdCallArgument).toEqual(1);
+
+            expect(createStub).toHaveBeenCalledOnce();
```

Comment:
Please try to add `toHaveBeenCalledWith(...)` in the future :+1:

Reply:
The parameters are checked in line 107-111.